webpack.common.js

```diff
@@ -16,46 +15,46 @@ module.exports = {
         'sva-finder': './src/Components/SvaFinder.js',
         'act-widget': './src/Components/ActWidget.js',
         'donation-widget': './src/Components/DonationWidget.js',
-        pdfworker: './src/Utility/PdfWorker.js',
         'test-interface': './src/test-interface.js',
         // We need to define a dummy entrypoint that requires all our translation files, otherwise Webpack will not
         // process them.
         'translations-dummy': ['./src/i18n/de.json', './src/i18n/en.json', './src/i18n/fr.json', './src/i18n/pt.json'],
     },
     optimization: {
-        minimize: true,
-        minimizer: [
-            new TerserPlugin({
-                sourceMap: true,
-                extractComments: false,
-                cache: true,
-                parallel: true,
-                terserOptions: {
-                    mangle: {
-                        reserved: [
-                            'ActionButtonPlaceholder',
-                            'NewRequestButtonPlaceholder',
-                            'CompanySelectorPlaceholder',
-                            'RequestFormPlaceholder',
-                            'DynamicInputContainerPlaceholder',
-                            'SignatureInputPlaceholder',
-                            'RequestTypeChooserPlaceholder',
-                            'RecipientInputPlaceholder',
-                            'TransportMediumChooserPlaceholder',
-                        ],
-                    },
+        splitChunks: {
+            cacheGroups: {
+                vendors: {
+                    name: 'vendors',
+                    chunks: 'all',
+                    test: /[/\\]node_modules[/\\]/,
+                    minChunks: 2,
                 },
-            }),
-        ],
+                commons: {
+                    name: 'commons',
+                    chunks: 'initial',
+                    minChunks: 2,
+                },
+            },
+        },
     },
     output: {
         filename: 'js/[name].gen.js',
         chunkFilename: 'js/[name].bundle.gen.js',
         publicPath: '/',
         path: path.resolve(__dirname, 'static'),
+        globalObject: 'self',
```

Comment:
This is the first time I am seeing this option, so I may well be wrong. But we have some code that depends on the 'raw' (untranspiled) JS added using Hugo and the Webpack-transpiled JS having access to the same `window` object. Is this option a good idea in this case?

Reply:
Although, now that I think about it some more, in those instances we are already explicitly using `window.something` instead of just `something`, so this may not apply.

Reply:
This was something I ran into debugging the loader